### PR TITLE
Redo Syntax.ebnf

### PR DESCRIPTION
**Pull request by rdrpenguin04**

### CODE REVIEW

File: docs/Syntax.ebnf [hunk 1]
```diff
@@ -40,6 +40,6 @@ codeblock = "{", {statement}, "}"; (* Block of code *)
 
 extern = "extern", ((funcdef, [":", type]) | ("{", [funcdef, [":", type], {"," funcdef, [":", type]}], "}")); (* Definition of external function(s) *)
 structure = "struct", identifier, "{", [definition, {"," definition}], "}"; (* Structure *)
-function = ["impl"], funcdef, (codeblock | ("->", expression)); (* Function *)
+function = ["impl"], funcdef, (("{", {statement}, [expression], "}") | ("->", expression, ";")); (* Function *)
```
Comment (jorolf) [inline docs/Syntax.ebnf]: every codeblock can have an implicit return...

Comment (rdrpenguin04) [reply docs/Syntax.ebnf]: Really? So I can have:
```
if (1 == 1) {
  2
}
```

File: docs/Syntax.ebnf [hunk 2]
```diff
@@ -1,62 +1,49 @@
...
+("=" | "//" | "/*" | "*/") (* Invalid or special *)
+; (* all operators *)
+
+type = identifier; (* Any type *)
+parameter = identifier + ":", type; (* Parameter to function *)
+
+comment = ("//", ? all characters ?, "\n") | ("/*", ? all characters ?, "*/");
+
+vardef = ("var" | "let"), identifier; (* Beginning of variable definition *)
+funcdef = "fn", identifier, "(", parameter, {",", parameter}, ")"; (* Beginning of function definition *)
```
Comment (jorolf) [inline docs/Syntax.ebnf]: parameters are optional

File: docs/Syntax.ebnf [hunk 3]
```diff
@@ -1,62 +1,49 @@
...
+opdef = "infix", "op", operator, "(", parameter, {",", parameter}, ")"; (* Beginning of function definition *)
```
Comment (jorolf) [inline docs/Syntax.ebnf]: infix isn't the only option, there also prefix and suffix and you can put the affixes on functions as well

File: docs/Syntax.ebnf [hunk 4]
```diff
@@ -1,62 +1,49 @@
...
+
+type = identifier; (* Any type *)
+parameter = identifier + ":", type; (* Parameter to function *)
+
+comment = ("//", ? all characters ?, "\n") | ("/*", ? all characters ?, "*/");
+
+vardef = ("var" | "let"), identifier; (* Beginning of variable definition *)
+funcdef = ["infix" | "suffix" | "prefix"], "fn", identifier, "(", parameter, {",", parameter}, ")"; (* Beginning of function definition *)
+opdef = ("infix" | "suffix" | "prefix"), "op", operator, "(", parameter, {",", parameter}, ")"; (* Beginning of function definition *)
+definition = (vardef | funcdef), [":", type]; (* Full definition of either variable or function *)
```
Comment (jorolf) [inline docs/Syntax.ebnf]: where is this used?

**Pull request merged**